ansible_base/lib/utils/views/urls.py

```diff
@@ -0,0 +1,29 @@
+from typing import Type, Union
+
+from django_extensions.management.commands.show_urls import Command as ShowUrlsCommand
+from rest_framework.views import APIView, View
+
+
+def get_view_functions(urlpatterns) -> set[Union[Type[APIView], Type[View]]]:
+    """
+    Extract view classes from a urlpatterns list using the show_urls helper functions
+
+    :param urlpatterns: django urlpatterns list
+    :return: set of all view classes used by the urlpatterns list
+    """
+    views = set()
+
+    # Get all active views
+    view_functions = ShowUrlsCommand().extract_views_from_urlpatterns(urlpatterns)
+    for func, _, _ in view_functions:
+        view = None
+        # ApiView.as_view() breadcrumb
+        if hasattr(func, 'cls'):
+            view = func.cls
+        if hasattr(func, "view_class"):
+            view = func.view_class
```

Comment:
Lets double check that you can't change the permissions classes for a specific view method. i.e. have [AllowAny] just on an options view.

Reply:
Testing it out, does not seem like the permission class decorator influences permissions for individual functions on `APIView`s

Reply:
My test:

```python
from rest_framework.decorators import permission_classes as mutate_permission_classes
# ...
class ApiRootView(AnsibleBaseView):
    permission_classes = (AllowAny,)

    @mutate_permission_classes([IsSuperuserOrAuditor])
    def get(self, request, format=None):
        # ...
        return Response(data)
```